### PR TITLE
Twitter Threads: Handle incorrectly formatted URLs when generating Twitter Cards

### LINKED ISSUE #18103: Fatal error when tweetstorm invoked with non-http url
On WordPress.org the following fatal error was encountered:
```
POST https://example.com/wp-json/wpcom/v2/tweetstorm/generate-cards?_locale=user

E_ERROR: Uncaught Requests_Exception: Only HTTP(S) requests are handled. in wp-includes/class-requests.php:566
Uncaught Requests_Exception: Only HTTP(S) requests are handled. in wp-includes/class-requests.php:566
Stack trace:
#0 wp-includes/class-requests.php(460): Requests::set_defaults('WordPress.org', Array, Array, 'GET', Array)
#1 wp-content/plugins/jetpack/_inc/lib/class-jetpack-tweetstorm-helper.php(1607): Requests::request_multiple(Array)
#2 wp-content/plugins/jetpack/_inc/lib/core-api/wpcom-endpoints/class-wpcom-rest-api-v2-endpoint-tweetstorm-parse.php(104): Jetpack_Tweetstorm_Helper::generate_cards(Array)
#3 wp-includes/rest-api/class-wp-rest-server.php(1160): WPCOM_REST_API_V2_Endpoint_Tweetstorm_Parse->generate_cards(Object(WP_REST_Request))
```

I'm unsure of the input to the endpoint that caused it, other than to say that it came from a post being edited, which doesn't appear to have any specific blocks related to Jetpack - https://wordpress.org/news/2020/12/introducing-learn-wordpress/   Revision history doesn't show any specific blocks, but at the time of the fatal the Youtube video was being inserted, it could also be related to the hashtag links included at the bottom?

It looks like Tweetstorm doesn't use WP_HTTP, but uses Requests directly with the multiple-request support, [it needs to either validate the URI's it's passing or catch exceptions (or both)](https://github.com/Automattic/jetpack/blob/9a437395f0d2f8248bea436faacd8445e71de846/_inc/lib/class-jetpack-tweetstorm-helper.php#L1597-L1607)

#### Steps to reproduce the issue

Unknown. 

#### What I expected

A valid API response.

#### What happened instead

A fatal error.

**Screenshots**

### PR DESCRIPTION
**Pull request by pento**
When generating Twitter Cards, we need to ensure that URLs are well formed before passing them to `Requests`.

Also, the return data previously used the `Requests`-formatted URL as the result key, which could cause the card to not be shown in the preview, if the URL was typed differently.

Fixes #18103.

#### Changes proposed in this Pull Request:
* Improve the URL validation and handling in `Jetpack_Tweetstorm_Helper::generate_cards()`.

#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:

* Create a new Twitter thread post.
* Type some weirdly formatted URLs in the post (eg, missing protocol, capitalisation).
* Check that there are no PHP errors in your error log.
* Check that the card shows correctly in the thread preview.

I've also added unit tests for this: `yarn docker:phpunit --filter twitter_card --group external-http`

#### Proposed changelog entry for your changes:
* Twitter Threads: Improve how Twitter Cards are generated for embeds in the thread preview.

### CODE REVIEW

**Review by jeherve — APPROVED** [review]

**Review by jeherve — APPROVED** [review]
Re-approving after merging the latest `master` to address unrelated test failures.

**Comment by pento** [comment]
r218502-wpcom

**Pull request merged**
